internal/template.go

```diff
@@ -13,29 +14,37 @@ import (
 	"k8s.io/helm/pkg/engine"
 )
 
+// WriteFileD dumps a given content on the file with path `targetDir/fileName`.
+func WriteFileD(fileName string, targetDir string, content string) error {
+	targetFilePath := targetDir + "/" + fileName
+	_ = os.Mkdir(targetDir, os.ModePerm)
+	return WriteFile(targetFilePath, content)
+}
+
+// WriteFile dumps a given content on the file with path `targetFilePath`.
+func WriteFile(targetFilePath string, content string) error {
+	return ioutil.WriteFile(targetFilePath, []byte(content), 0777)
+}
+
 // ExecuteTemplate loads a template file, executes is against a given function map and writes the output
-func ExecuteTemplate(sourceFilePath string, funcMap template.FuncMap, verbose bool, dryRun bool) error {
+func ExecuteTemplate(sourceFilePath string, funcMap template.FuncMap, verbose bool) (string, error) {
```

Comment:
I would do the other way around, keep the logic agnostic and continue to work on strings and handle the writing to file in a separate section (for now it can be simply the main).